### PR TITLE
Remove command delay

### PR DESCRIPTION
**Pull request by zeozeozeo**
Fixes command delay by processing pending commands for each audio frame. Fixes https://github.com/tesselode/kira/issues/65 (and includes https://github.com/tesselode/kira/pull/69)
This will fix issues such as volume commands not applying immediately and will remove the delay between the `stream.play()` call and the actual sound being played.

This shouldn't add much overhead (I haven't benchmarked it though). You can pass the `process_command_interval` option to set the amount of audio frames between `on_start_processing` calls. Currently set to 1 (NonZeroUsize), but it might be better to increase that value (needs benchmarking).

### CODE REVIEW

File: crates/kira/src/manager/backend/cpal/desktop/stream_manager.rs [hunk 1]
```diff
@@ -197,10 +213,17 @@ fn device_name(device: &Device) -> String {
...
-		// process pending commands
-		renderer_wrapper.on_start_processing();
+fn process_renderer(
+	renderer_wrapper: &mut RendererWrapper,
+	data: &mut [f32],
+	channels: u16,
+	process_command_interval: usize,
+) {
+	for (i, frame) in data.chunks_exact_mut(channels as usize).enumerate() {
+		if i % process_command_interval == 0 {
```
Comment (zeozeozeo) [inline crates/kira/src/manager/backend/cpal/desktop/stream_manager.rs:223]: this if condition will run even if it is the first frame, because 0 % n = 0

**Comment by tesselode** [comment]
Do you have a way to test the delay between the gameplay thread sending commands and the audio thread receiving them?

**Comment by zeozeozeo** [comment]
> Do you have a way to test the delay between the gameplay thread sending commands and the audio thread receiving them?

I thought the command list was something like a vector under a Mutex, so it should have no delay between updating in the audio thread? I would think that the only delay that could be present is because that Kira only processes the command list once before it starts rendering the audio samples, so the next command is only processed in the next buffer given by cpal, which is only processed when the previous buffer starts playing

**Comment by tesselode** [comment]
I think you're going to find that your change doesn't change the latency much, since Kira processes all the samples in a buffer as fast as possible, so unless those samples are very CPU-intensive, they're all going to process in a short period of time, and then the thread will sleep until the next buffer period.

**Comment by zeozeozeo** [comment]
yeah, you're right, not sure how i didn't think about that

**Pull request closed without merging**